mtp_api/apps/transaction/views.py

```diff
@@ -52,9 +54,87 @@ class TransactionView(
 ):
     queryset = Transaction.objects.all()
     serializer_class = TransactionSerializer
+    patch_serializer_class = CreditedOnlyTransactionSerializer
     filter_backends = (
         filters.DjangoFilterBackend,
         filters.OrderingFilter,
     )
-    filter_class = TransactionFilter
+
+    filter_class = TransactionStatusFilter
+
     ordering = ('received_at',)
+    permission_classes = (IsAuthenticated, IsOwnPrison,)
+
+    def get_queryset(self, filter_by_user=True, filter_by_prison=True):
+        qs = super(TransactionView, self).get_queryset()
+
+        prison_id = self.kwargs.get('prison_id')
+        user_id = self.kwargs.get('user_id')
+
+        if prison_id and filter_by_prison:
+            qs = qs.filter(prison_id=prison_id)
+
+        if user_id and filter_by_user:
+            qs = qs.filter(owner__id=user_id)
+
+        return qs
+
+    def take(self, request, *args, **kwargs):
```

Comment:
The more I think about it, the more I think this should be `assign` rather than `take`. Waddya think?